deployment/data-streams/changeset/call_configurator_set_staging_config.go

```diff
@@ -0,0 +1,172 @@
+	OnchainConfig         []byte
+	OffchainConfigVersion uint64
+	OffchainConfig        []byte
+}
+
+func (sc SetStagingConfig) GetConfiguratorAddress() common.Address {
+	return sc.ConfiguratorAddress
+}
+
+// Validate checks the staging configuration.
+func (cfg SetStagingConfigConfig) Validate() error {
+	if len(cfg.ConfigurationsByChain) == 0 {
+		return errors.New("ConfigurationsByChain cannot be empty")
+	}
+	return nil
+}
+
+func callSetConfigCommon[T ConfiguratorConfig](
+	e deployment.Environment,
+	configurationsByChain map[uint64][]T,
+	mcmsConfig *MCMSConfig,
+	proposalName string,
+	txBuilder func(e deployment.Environment, configuratorContract *configurator.Configurator, cfg T, opts *bind.TransactOpts, chain deployment.Chain, mcmsConfig *MCMSConfig) (*ethTypes.Transaction, error),
+) (deployment.ChangesetOutput, error) {
+	// Collect chain selectors.
+	chainSelectors := make([]uint64, 0, len(configurationsByChain))
+	for chainSelector := range configurationsByChain {
+		chainSelectors = append(chainSelectors, chainSelector)
+	}
+
+	var (
+		mcmsStatePerChain map[uint64]*changeset.MCMSWithTimelockState
+		err               error
+	)
+	if mcmsConfig != nil {
+		mcmsStatePerChain, err = changeset.MaybeLoadMCMSWithTimelockState(e, chainSelectors)
+		if err != nil {
+			return deployment.ChangesetOutput{}, err
+		}
+	}
+
+	mcmsPerChain := map[uint64]*owner_helpers.ManyChainMultiSig{}
+	timelockAddresses := map[uint64]common.Address{}
+	allBatches := []timelock.BatchChainOperation{}
+
+	// Process each chain's configurations.
+	for chainSelector, configs := range configurationsByChain {
+		chain, ok := e.Chains[chainSelector]
+		if !ok {
+			return deployment.ChangesetOutput{}, fmt.Errorf("chain %d not found in environment", chainSelector)
+		}
+		chainID := mcms.ChainIdentifier(chainSelector)
+		batch := timelock.BatchChainOperation{
+			ChainIdentifier: chainID,
+			Batch:           []mcms.Operation{},
+		}
+
+		if mcmsConfig != nil {
+			mcmsState := mcmsStatePerChain[chainSelector]
+			tlAddress := mcmsState.Timelock.Address()
+			mcmsPerChain[uint64(chainID)] = mcmsState.ProposerMcm
+			timelockAddresses[chainSelector] = tlAddress
+		}
+
+		opts := getTransactOpts(e, chainSelector, mcmsConfig)
+		for _, cfg := range configs {
+			state, err := maybeLoadConfigurator(e, chainSelector, cfg.GetConfiguratorAddress().String())
+			if err != nil {
+				return deployment.ChangesetOutput{}, err
+			}
+
+			tx, err := txBuilder(e, state.Configurator, cfg, opts, chain, mcmsConfig)
+			if err != nil {
+				return deployment.ChangesetOutput{}, err
+			}
+
+			op := mcms.Operation{
+				To:           cfg.GetConfiguratorAddress(),
+				Data:         tx.Data(),
+				Value:        big.NewInt(0),
+				ContractType: string(types.Configurator),
+			}
+			batch.Batch = append(batch.Batch, op)
+		}
+
+		allBatches = append(allBatches, batch)
+	}
+
+	if mcmsConfig != nil {
+		proposal, err := proposalutils.BuildProposalFromBatches(
```

Comment:
Should we be using BuildProposalFromBatchesV2 - this func `BuildProposalFromBatches` says it's deprecated